orange_tree2.rb

```diff
@@ -0,0 +1,108 @@
+class OrangeTree
+  attr_reader :height, :age, :orange_count, :death_age
+
+  # creates new tree
+  def initialize
+    @height = 1
+    @age = 0
+    @orange_count = 0
+    #random age between 90-110
+    @death_age = (rand(90..110))
+    @max_height = 30
+  end
+
+  def increases_height
+    #increases height according to age.
+    if (@age <= 5)
+      @height += 3
+      #puts "#{@height}"
+    elsif (@age <= 15)
+      @height += 1
+    elsif (@age <= 40) && (@height < @max_height)
+      @height += 0.5
+    end
+  end
+
+  def produces_oranges
+    #produces orange? according to age
+    if (@age < 6)
+      @orange_count = 0
+    elsif (@age >= 6) && (@age < 20)
+      @orange_count = (rand(50..100))
+    elsif (@age >= 20) && (@age < 40)
+      @orange_count = (rand(100..200))
+    elsif (@age >= 40) && (@age < 60)
+      @orange_count = (rand(200..300))
+    else
+      @orange_count = (rand(100..200))
+    end
+  end
+
+  def one_year_passes
+    #increases age by one year
+    @age += 1
+    increases_height
+    produces_oranges
+    #is alive?
+    if (@age >= @death_age)
+      puts "Tree has passed on. Good-bye, tree."
+    end
+  end
+
+  def pick_an_orange
+    @orange_count -= 1
+    if (@orange_count <= 0)
+      puts "Sorry, tree is empty!"
```

Comment:
The assignment says that this function should return a string.

If the method returns the string, then other code can call the method, get the result, and do something with it.

Here is some code that you can paste into irb to see the difference:
```ruby
def puts_3
  puts 3 # prints to the console, but returns nil - note: nil is like "nothing" and is not the same as zero
end

def return_3
  3 # same as `return 3` - in ruby, "return" is magical this way
end
```

Then you can try things like this:

```ruby
a = puts_3 # 3 is printed, but a is now nil
b = return_3 # b is now 3
puts b * b # prints 9
puts a * a # raises a NoMethodError because nil doesn't know how to multiply
```

Hope this helps!